creator-node/test/audiusUsers.js

```diff
@@ -66,3 +85,91 @@ describe('test AudiusUsers', function () {
       .expect(200)
   })
 })
+
+// Below block uses actual ipfsClient (unlike first describe block), hence
+// another describe block for this purpose
+describe('tests /audius_users/metadata metadata upload with actual ipfsClient', function () {
+  let app, server, session, libsMock, ipfs
+
+  // Will need a '.' in front of storagePath to look at current dir
+  // a '/' will search the root dir
+  before(async () => {
+    let storagePath = config.get('storagePath')
+    if (storagePath.startsWith('/')) {
+      storagePath = '.' + storagePath
+      config.set('storagePath', storagePath)
+    }
+  })
+
+  beforeEach(async () => {
+    ipfs = ipfsClient.ipfs
+    libsMock = getLibsMock()
+
+    const appInfo = await getApp(ipfs, libsMock, BlacklistManager)
+    await BlacklistManager.blacklist(ipfs)
+
+    app = appInfo.app
+    server = appInfo.server
+    session = await createStarterCNodeUser()
+  })
+
+  afterEach(async () => {
+    sinon.restore()
+    await server.close()
+  })
+
+  it('should fail if metadatda is not found in request body', async function () {
+    const resp = await request(app)
+      .post('/audius_users/metadata')
+      .set('X-Session-ID', session)
+      .send({ dummy: 'data' })
+      .expect(500)
+
+    assert.deepStrictEqual(resp.body.error, 'Internal server error')
+  })
+
+  it('should throw error response if saving metadata to fails', async function () {
+    sinon.stub(ipfs, 'add').rejects(new Error('ipfs add failed!'))
```

Comment:
isn't this describe block to test with real ipfsClient? seems that this test should either be moved to first describe block or be modified to temporarily induce ipfs unavailability

Reply:
ah, my intention was that for the happy paths where IPFS is working, use the actual IPFS client. otherwise, it's okay to mock an error being thrown. but you're right, this scenario should be included in the description. will add